### PR TITLE
Android IPC Refactor

### PR DESCRIPTION
**Pull request by jwerle**

### CODE REVIEW

File: src/core/json.cc [hunk 1]
```diff
@@ -5,6 +5,72 @@ namespace SSC::JSON {
...
+      stream << value;
+
+      if (--count > 0) {
+        stream << std::string(",");
+      }
+    }
+
+    stream << std::string("}");
+    return stream.str();
+  }
```
Comment (chicoxyzzy) [inline src/core/json.cc:51]: hmm... I think this was working already?

Comment (jwerle) [reply src/core/json.cc:51]: yeah it was - I just moved the implementation into `json.cc`

**Review by chicoxyzzy — COMMENTED** [review]
I think we can rename `src/mobile/` to `src/ios` now

**Comment by jwerle** [comment]
> I think we can rename `src/mobile/` to `src/ios` now

Yeah it would be nice to also clean up and consolidate the iOS stuff too

**Review by chicoxyzzy — COMMENTED** [review]
if we drop `realpath` everywhere, we also can drop installing `coreutils` in https://github.com/socketsupply/socket/blob/ipc-cleanup/.github/workflows/ci.yml (on lines 208 and 255)

**Comment by jwerle** [comment]
> if we drop `realpath` everywhere, we also can drop installing `coreutils` in https://github.com/socketsupply/socket/blob/ipc-cleanup/.github/workflows/ci.yml (on lines 208 and 255)

dropped 🤓

**Pull request merged**